src/vstest.console/Processors/TestAdapterPathArgumentProcessor.cs

```diff
@@ -155,63 +155,55 @@ public void Initialize(string argument)
+                    var testAdapterFullPaths = new List<string>();
                     var testAdapterPathsInRunSettingsArray = testAdapterPathsInRunSettings.Split(
                         new[] { ';' },
                         StringSplitOptions.RemoveEmptyEntries);
 
-                    if (testAdapterPathsInRunSettingsArray.Length > 0)
+                    foreach (var testadapterPath in testAdapterPathsInRunSettingsArray)
                     {
-                        testadapterPaths.AddRange(testAdapterPathsInRunSettingsArray);
-                    }
-                }
+                        var testAdapterFullPath = Path.GetFullPath(testadapterPath);
 
-                foreach (var testadapterPath in testadapterPaths)
-                {
-                    var customAdaptersPath = Path.GetFullPath(testadapterPath);
+                        if (!this.fileHelper.DirectoryExists(testAdapterFullPath))
+                        {
+                            throw new DirectoryNotFoundException(CommandLineResources.TestAdapterPathDoesNotExist);
+                        }
 
-                    if (!this.fileHelper.DirectoryExists(customAdaptersPath))
-                    {
-                        throw new DirectoryNotFoundException(CommandLineResources.TestAdapterPathDoesNotExist);
+                        testAdapterFullPaths.Add(testAdapterFullPath);
                     }
 
-                    fullPathTestAdapterPaths.Add(customAdaptersPath);
+                    testAdapterFullPaths.Add(customAdaptersPath);
+                    testAdapterFullPaths = testAdapterFullPaths.Distinct().ToList();
+                    customAdaptersPath = string.Join(";", testAdapterFullPaths.ToArray());
                 }
 
-                fullPathTestAdapterPaths = fullPathTestAdapterPaths.Distinct().ToList();
-
-                fullPathTestAdapterPathsString = string.Join(";", fullPathTestAdapterPaths.ToArray());
-
-                this.runSettingsManager.UpdateRunSettingsNode("RunConfiguration.TestAdaptersPaths", fullPathTestAdapterPathsString);
+                this.runSettingsManager.UpdateRunSettingsNode("RunConfiguration.TestAdaptersPaths", customAdaptersPath);
             }
             catch (Exception e)
             {
                 throw new CommandLineException(
                     string.Format(CultureInfo.CurrentCulture, CommandLineResources.InvalidTestAdapterPathCommand, argument, e.Message));
             }
 
-            this.commandLineOptions.TestAdapterPath = fullPathTestAdapterPathsString;
-
-            if (fullPathTestAdapterPaths.Count <= 0)
-            {
-                // Print a warning about not finding any test adapter in provided path...
-                this.output.Warning(CommandLineResources.NoAdaptersFoundInTestAdapterPath, argument);
-                this.output.WriteLine(string.Empty, OutputLevel.Information);
-            }
+            this.commandLineOptions.TestAdapterPath = customAdaptersPath;
```

Comment:
Can we please check uses of this. earlier it were taking single path but now u are passing list of paths